### ISSUE TITLE
How to assign weights to AP?

### ISSUE THREAD

**Issue by fyang93** [naver/deep-image-retrieval#16]
As described in the paper, a weight is introduced into the calculation of mAP in order to counter-balance the dataset imbalance issue.
I guess the weight is simply inversely proportional to the number of samples of each class in a single batch, so I wrote code like
```python
uni_labels, inv_ids, counts = torch.unique(img_labels, return_inverse=True, 
                                           return_counts=True)
weights = 1 / counts.float()
weights = weights[inv_ids]  # query weights
...
loss = APLoss(scores, labels, weights)
```
But training by this code only results in explosion in the loss, while anything works fine without the weight.
Could you give an example on how to assign the weights to AP?
Thank you very much.
